### PR TITLE
Workflow support

### PR DESCRIPTION
**Pull request by koparasy**
* Implements `orchestrator` to push sub-selection jobs and training jobs using flux
* Extends `stager` to receive model update notifications and push them to the user provided filter mechanism
* Initial support for a `workflow` description language
* Adds new example of binomial options, the benchmark can scale the time on a per element basis.
* Adds a training script (Thank you @ZwFink) for binomial

### CODE REVIEW

**Comment by koparasy** [comment]
@lpottier  this is 'ready' for review. I will start cleaning up the code and adding comments etc. Keep in mind that this is the first implementation of the end-to-end so the process is likely to be sub-optimal.

File: examples/tools/train.py [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+from ams.store_types import UQType
+from ams.store_types import UQAggregate
+from ams.views import AMSDataView
+import argparse
+from pathlib import Path
+import torch.nn.init as init
+
+
+class ExampleDataset(Dataset):
+    def __init__(self, X, y):
```
Comment (lpottier) [inline examples/tools/train.py]: Mixing lowercase and uppercase?

File: src/AMSWorkflow/ams/ams_flux.py [hunk 2]
```diff
@@ -0,0 +1,529 @@
...
+    * The ``uri`` and ``add_uri_callback`` methods for retrieving the
+      Flux uri of the instance executing this job. This is convenient for
+      nested jobs
+
+    Valid events are contained in the ``EVENTS`` class attribute.
+    """
+
+    # NOTE: This is the primary difference of the original FluxExecutorFuture.
+    # FluxExecutorFuture uses frozensets without adding "memo" and requires all registered
+    # callbacks  to be part of the EVENTS. Thus we cannot inherit directly from it and
```
Comment (lpottier) [inline src/AMSWorkflow/ams/ams_flux.py]: unfinished comment?

**Review by lpottier — APPROVED** [review]
Very big PR, looks good to me! Thanks @koparasy

**Pull request merged**